src/Simulation/Microsoft.Quantum.TargetGeneration/GenerateTarget.fs

```diff
@@ -0,0 +1,398 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Quantum.QsCompiler.TargetGeneration
+
+open Microsoft.CodeAnalysis.CSharp.Syntax
+
+open Microsoft.Quantum.RoslynWrapper
+
+open System.Collections.Generic
+open System.IO
+open Microsoft.Quantum.QsCompiler.CsharpGeneration
+open Microsoft.Quantum.QsCompiler.SyntaxTree
+open Microsoft.Quantum.QsCompiler.SyntaxTokens
+open Microsoft.Quantum.QsCompiler
+open Microsoft.Quantum.QsCompiler.SyntaxExtensions
+open Microsoft.Quantum.QsCompiler.DataTypes
+open System.Collections.Immutable
+open Microsoft.CodeAnalysis
+open Microsoft.CodeAnalysis.CSharp
+
+module GenerateTarget =
+    let private autoNamespaces =
+        [
+            "System"
+            "Microsoft.Quantum.Core"
```

Comment:
I just pulled this from the C# code gen. Andres, below, suggests instead using the fully-qualified names and not opening these; I think that's pretty straightforward to do, but I haven't done it yet. I know Intrinsics is referenced at least once.

Reply:
Is the intrinsics namespace is referenced for the qubit management?

Reply:
I've noticed this in the generated C# as well. The Core and Intrinsic namespaces are always added, even when not used. This doesn't match up with the Q# behavior, where at least the Intrinsic namespace as to be explicitly opened. I like Andres' idea of switching those to be fully-qualified names when called to avoid having these in the `autoNamespaces` list, and I think we should consider doing the same for the C# generation as well. 

Should I file an issue for that?

Reply:
@swernli That would be great -- that way we can merge this PR and address that issue later.